tests/test_sonify.py

```diff
@@ -53,3 +54,37 @@ def test_chords():
             ['C', 'C:maj', 'D:min7', 'E:min', 'C#', 'C', 'C', 'C', 'C', 'C'],
             intervals, fs, length=fs*11)
         assert len(signal) == 11*fs
+
+
+def test_pitch_contour():
+
+    # Generate some random pitch
+    fs = 8000
+    times = np.linspace(0, 5, num=5 * fs, endpoint=True)
+
+    noise = scipy.ndimage.gaussian_filter1d(np.random.randn(len(times)),
+                                            sigma=256)
+    freqs = 440.0 * 2.0**(16 * noise)
+
+    # negate a bunch of sequences
+    idx = np.unique(np.random.randint(0, high=len(times), size=32))
+    for start, end in zip(idx[::2], idx[1::2]):
+        freqs[start:end] *= -1
+
+    # Test with inferring duration
+    x = mir_eval.sonify.pitch_contour(times, freqs, fs)
+    assert len(x) == fs * 5
+
+    # Test with an explicit duration
+    # This forces the interpolator to go off the end of the sampling grid,
+    # which should put zeros at the end of the signal
```

Comment:
If by "signal" we mean "instatnaneous frequencies" then yes; I can clarify though.